### PR TITLE
Form input file

### PR DESCRIPTION
**Pull request by umeshnevase**

### CODE REVIEW

File: app/admin/RTMediaFormHandler.php [hunk 1]
```diff
@@ -244,6 +244,36 @@ public static function textbox( $args ) {
...
+	 * @access static
+	 *
+	 * @param  array $args
+	 *
+	 * @return void
+	 */
+	public static function inputfile( $args ) {
+		global $rtmedia;
+		$options = $rtmedia->options;
+		$defaults = array(
```
Comment (thrijith) [inline app/admin/RTMediaFormHandler.php]: equals sign alignment issue

File: app/admin/RTMediaFormHandler.php [hunk 2]
```diff
@@ -244,6 +244,36 @@ public static function textbox( $args ) {
...
+			'key' => '',
+			'desc' => '',
```
Comment (thrijith) [inline app/admin/RTMediaFormHandler.php]: double arrow not aligned properly

File: app/admin/RTMediaFormHandler.php [hunk 3]
```diff
@@ -244,6 +244,36 @@ public static function textbox( $args ) {
 		$numObj->display_textbox( $args );
 	}
 
+	/**
+	 * Show rtmedia input file in admin options.
+	 *
+	 * @access static
```
Comment (thrijith) [inline app/admin/RTMediaFormHandler.php]: `@access static` ?

File: app/helper/rtForm.php [hunk 4]
```diff
@@ -588,6 +645,25 @@ public function display_textbox( $args = '' ) {
...
 
+		/**
+		 * Get rtmedia html input file in admin options.
+		 *
+		 * @access public
+		 *
+		 * @param string/array $attributes
+		 *
+		 * @return string
+		 * @throws rtFormInvalidArgumentsException
```
Comment (thrijith) [inline app/helper/rtForm.php]: usage of `@throws` not in function

**Comment by umeshnevase** [comment]
Fixed feedback

File: app/helper/rtForm.php [hunk 5]
```diff
@@ -569,6 +572,60 @@ protected function generate_textbox( $attributes ) {
...
+		 */
+		protected function generate_inputfile( $attributes ) {
+
+			$element = 'rtFile';
+			if ( is_array( $attributes ) ) {
+
+				/* Starting the input tag */
+				$html = '<input type="file" ';
```
Comment (milindmore22) [inline app/helper/rtForm.php]: check if can add html in a file template

File: app/helper/rtForm.php [hunk 6]
```diff
@@ -569,6 +572,60 @@ protected function generate_textbox( $attributes ) {
...
+
+				/* Generating attributes */
+				$html .= $this->processAttributes( $element, $attributes );
+
+				/* Ending the tag */
+				$html .= ' /><input type="hidden" name="rtmedia-options[' . $attributes['name'] . '_hid]" value="' . $attributes['value'] . '" />';
```
Comment (milindmore22) [inline app/helper/rtForm.php]: use proper escaping

File: app/helper/rtForm.php [hunk 7]
```diff
@@ -569,6 +572,60 @@ protected function generate_textbox( $attributes ) {
...
+				if ( ! empty( $attributes['value'] ) ) {
+					$img_src  = wp_get_attachment_image_src( $attributes['value'], 'thumbnail' );
+					$img_path = get_attached_file( $attributes['value'] );
+
+					if ( file_exists( $img_path ) && ! empty( $img_src[0] ) ) {
+						$html .= '<span class="rtm-file-preview">';
+						$html .= '<img src="' . $img_src[0] . '" width="100">';
```
Comment (milindmore22) [inline app/helper/rtForm.php]: use proper escaping always process string with a variable in `sprintf`
eg: 
``` $html .= ''; ```
to
``` $html .= sprintf( ', esc_url( $img_src[0] ) ); ```

File: tests/bootstrap.php [hunk 8]
```diff
@@ -1,20 +0,0 @@
-<?php
```
Comment (milindmore22) [inline tests/bootstrap.php]: why removed  ?

File: app/helper/rtForm.php (hunk 5 above)
Comment (umeshnevase) [reply app/helper/rtForm.php]: Will make lots of changes to existing input fields as well, we need to convert all input fields functions to html template

File: app/helper/rtForm.php [hunk 9]
```diff
@@ -569,6 +572,61 @@ protected function generate_textbox( $attributes ) {
...
+				/* Ending the tag */
+				$html .= ' /><input type="hidden" name="rtmedia-options[' . esc_attr( $attributes['name'] ) . '_hid]" value="' . esc_attr( $attributes['value'] ) . '" />';
+				if ( ! empty( $attributes['value'] ) ) {
+					$img_src  = wp_get_attachment_image_src( $attributes['value'], 'thumbnail' );
+					$img_path = get_attached_file( $attributes['value'] );
+
+					if ( file_exists( $img_path ) && ! empty( $img_src[0] ) ) {
+						$html .= '<span class="rtm-file-preview">';
+						$html .= sprintf( '<img src="%s" width="100">', esc_url( $img_src[0] ) );
+						$html .= '<a href="#" class="no-popup rtm-delete-preview" title="' . esc_attr__( 'Delete this file', 'buddypress-media' ) . '" data-media_type="' . $attributes['name'] . '">';
```
Comment (chandrapatel) [inline app/helper/rtForm.php]: @umeshnevase Please escape `$attributes['name']` using `esc_attr()`

**Comment by VijayaRaghavanM** [comment]
Got confirmation from Umesh to close the PR.

**Pull request closed without merging**